app.ts

```diff
@@ -45,11 +46,11 @@ app.post('/replicache-batch', async (req, res) => {
 });
 
 function handleError(res, e) {
-    if (e instanceof UnauthorizedError) {
+    if (e.kind == ErrorKind.Unauthorized) {
         res.status(status.UNAUTHORIZED).send(e.toString());
         return;
     }
-    if (e instanceof UserError) {
+    if (e.kind == ErrorKind.UserError) {
```

Comment:
@arv when I made the change to typescript, this instanceof stopped working in one specific case. I manually confirmed that gcal.js was really `throw new UserError()`. My best guess is nodejs/node#13408 somehow, but couldn't figure it out. Did this as a workaround.

Reply:
Looking at the code it does not seem like the ordering is an issue here since `errors.ts` has no imports so there is no risk of ordering issues.

More likely due to `extends Error` and compiling to ES5. (I haven't checked what tsconfig you are using yet)

Reply:
If you want to stick to property checking Error objects have a `name` that matches the class name.